src/Simulations/run.jl

```diff
@@ -189,5 +189,8 @@ function run!(sim; pickup=false)
         sim.run_time += time_after - time_before
     end
 
+    # Checkpoint at the end of `run!`
+    [write_output!(writer, sim.model) for writer in values(sim.output_writers) if writer isa Checkpointer]
```

Comment:
Is there a use case for multiple checkpointers?

Reply:
I agree this is useful! And I also agree that we probably don't wanna do that by default. 

In my opinion multiple checkpointers would be an unnecessary extra complexity to a simulation. I'd just add a keyword like `write_at_end=false` to `Checkpointer`, or something along those lines.

Reply:
Potentially. I can imagine using two checkpointers, one for actually picking up from on subsequent Slurm jobs where I only keep the latest checkpoint, and a second one for saving the 3D state of the model where I keep all of them. Perhaps better achieved by a 3D output writer, but why stop users from using multiple checkpoints.

Right now we can't assume that there's only one `Checkpointer` so we have to checkpoint all of them.